libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -35,6 +39,8 @@ export class BookSearchComponent implements OnInit {
     this.store.select(getAllBooks).subscribe(books => {
       this.books = books;
     });
+
+    this.triggerInstantSearch.pipe(auditTime(500)).subscribe(() => this.searchBooks());
```

Comment:
You might want to rethink this. `auditTime` isn't really the right operator for a `debounce` as it will just emit every 500ms even if the input is changing. What you want is to wait for the value to stop changing for 500ms and then emit. There's an operator that does this...

Reply:
You're also not unsubscribing from this, which may not be a big deal for this `Subject`, but is good practice.

Reply:
I made this change. I was under the impression we wanted to emit as the value was changing.